tests/main/refresh/task.yaml

```diff
@@ -99,33 +102,34 @@ execute: |
     # echo "================================="
 
     echo "When the snap is refreshed"
-    snap refresh --channel=edge $SNAP_NAME
+    snap refresh --channel=edge "$SNAP_NAME"
 
     echo "Then the new version is listed"
     expected="$SNAP_NAME +$SNAP_VERSION_PATTERN"
     snap list | grep -Pzq "$expected"
 
     echo "When a snap is refreshed and has no update it exit 0"
-    snap refresh $SNAP_NAME 2>stderr.out
-    cat stderr.out | MATCH "snap \"$SNAP_NAME\" has no updates available"
+    snap refresh "$SNAP_NAME" 2>stderr.out
+    MATCH "snap \"$SNAP_NAME\" has no updates available" < stderr.out
 
     echo "classic snaps "
 
     echo "When multiple snaps have no update we have a good message"
-    . $TESTSLIB/snaps.sh
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
     install_local basic
-    snap refresh $SNAP_NAME basic 2>&1 | MATCH "All snaps up to date."
+    snap refresh "$SNAP_NAME" basic 2>&1 | MATCH "All snaps up to date."
 
     echo "When moving to stable"
-    snap refresh --stable $SNAP_NAME
-    snap info $SNAP_NAME | MATCH "tracking: +stable"
+    snap refresh --stable "$SNAP_NAME"
+    snap info "$SNAP_NAME" | MATCH "tracking: +stable"
 
-    snap refresh --candidate $SNAP_NAME 2>&1 | MATCH "$SNAP_NAME \(candidate\).*"
-    snap info $SNAP_NAME | MATCH "tracking: +candidate"
+    snap refresh --candidate "$SNAP_NAME" 2>&1 | MATCH "$SNAP_NAME \\(candidate\\).*"
+    snap info "$SNAP_NAME" | MATCH "tracking: +candidate"
 
     echo "When multiple snaps are refreshed we error if we have unknown names"
     if snap refresh core invälid-snap-name 2> out.err; then
         echo "snap refresh invalid-snap-name should fail but it did not?"
         exit 1
     fi
-    cat out.err | tr '\n' ' ' | tr -s ' ' | MATCH 'cannot refresh .* is not installed'
+    < out.err tr '\n' ' ' | tr -s ' ' | MATCH 'cannot refresh .* is not installed'
```

Comment:
I'm not sure about this one - this looks quite unusual maybe allowing cat in this specific case is ok? Or maybe we can find something else to make this easier to read?

Reply:
Hmm some alternatives:
```
tr '\n' ' ' < out.err | tr -s ' ' | MATCH 'cannot refresh .* is not installed'
(tr '\n' ' ' | tr -s ' ' | MATCH 'cannot refresh .* is not installed') < out.err 
```
Or I could just disable the warning:
```
# shellcheck disable=SC2002
cat out.err | tr ...
```

Reply:
I (slightly) prefer to simply allow cat in this case, its not like its terrible and I think in this case its slightly easier to read.